pkg/controller/installation/installation_controller.go

```diff
@@ -154,13 +156,18 @@ func createInstallationCR(ctx context.Context, serverClient k8sclient.Client) er
 		useClusterStorage, _ := os.LookupEnv("USE_CLUSTER_STORAGE")
 		alertingEmailAddress, _ := os.LookupEnv(alertingEmailAddressEnvName)
 		installType, _ := os.LookupEnv(installTypeEnvName)
+		priorityClassName, _ := os.LookupEnv(priorityClassNameEnvName)
 
 		logrus.Infof("Creating a %s rhmi CR with USC %s, as no CR rhmis were found in %s namespace", installType, useClusterStorage, namespace)
 
 		if installType == "" {
 			installType = string(integreatlyv1alpha1.InstallationTypeManaged)
 		}
 
+		if installType == string(integreatlyv1alpha1.InstallationTypeManagedApi) && priorityClassName == "" {
+			priorityClassName = managedServicePriorityClassName
+		}
```

Comment:
I see this block is reached only when the CR does not exist. When upgrading the operator from a previous version, as the installation is already complete and the CR exists, the `.spec.priorityClassName` field won't be set automatically, so the PriorityClass won't be created unless the field is set manually. If it's something that we want to support maybe we could move this logic to the Reconcile loop